diff --git a/openstack/network/v2/security_group_rule.py b/openstack/network/v2/security_group_rule.py
index 06de5b3b3..6653ab923 100644
--- a/openstack/network/v2/security_group_rule.py
+++ b/openstack/network/v2/security_group_rule.py
@@ -27,7 +27,7 @@ class SecurityGroupRule(_base.NetworkResource, resource.TagMixin):
     allow_list = True
 
     _query_mapping = resource.QueryParameters(
-        'description', 'direction', 'protocol',
+        'description', 'direction', 'id', 'protocol',
         'remote_group_id', 'security_group_id',
         'remote_address_group_id',
         'port_range_max', 'port_range_min',
diff --git a/openstack/tests/unit/network/v2/test_security_group_rule.py b/openstack/tests/unit/network/v2/test_security_group_rule.py
index 660883e58..c334b654e 100644
--- a/openstack/tests/unit/network/v2/test_security_group_rule.py
+++ b/openstack/tests/unit/network/v2/test_security_group_rule.py
@@ -51,6 +51,7 @@ def test_basic(self):
         self.assertDictEqual({'any_tags': 'tags-any',
                               'description': 'description',
                               'direction': 'direction',
+                              'id': 'id',
                               'ether_type': 'ethertype',
                               'limit': 'limit',
                               'marker': 'marker',
